java/app/src/main/java/org/vss/auth/JwtAuthorizer.java

```diff
@@ -0,0 +1,69 @@
+package org.vss.auth;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.JWTVerifier;
+import jakarta.ws.rs.core.HttpHeaders;
+import org.vss.exception.AuthException;
+
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+public class JwtAuthorizer implements Authorizer {
+
+	private final PublicKey publicKey;
+	private final JWTVerifier verifier;
+
+	public JwtAuthorizer(String pemFormatPublicKey) throws Exception {
+		this.publicKey = loadPublicKey(pemFormatPublicKey);
+
+		Algorithm algorithm = Algorithm.RSA256((RSAPublicKey) publicKey, null);
+		this.verifier = JWT.require(algorithm).build();
+	}
+
+	@Override
+	public AuthResponse verify(HttpHeaders headers) throws AuthException {
+
+		try {
+			String authorizationHeader = headers.getHeaderString(HttpHeaders.AUTHORIZATION);
+			if (authorizationHeader == null || !authorizationHeader.startsWith("Bearer ")) {
+				throw new AuthException("Missing or invalid Authorization header.");
+			}
+
+			// Extract token by excluding 'Bearer '.
+			String token = authorizationHeader.substring(7);
+
+			DecodedJWT jwt = verifier.verify(token);
+
+			// Extract the user identity from the token.
+			String userToken = jwt.getSubject();
+
+			if (userToken == null || userToken.isBlank()) {
+				throw new AuthException("Invalid JWT token.");
+			}
+
+			return new AuthResponse(userToken);
+
+		} catch (JWTVerificationException e) {
+			throw new AuthException("Invalid JWT token.");
+		}
+	}
+
+	private PublicKey loadPublicKey(String pemFormatPublicKey) throws Exception {
+		String key = pemFormatPublicKey
+				.replaceAll("\\n", "")
+				.replace("-----BEGIN PUBLIC KEY-----", "")
+				.replace("-----END PUBLIC KEY-----", "");
```

Comment:
Could there be other sections in the file not corresponding to the public key? Should we take what's between the markers instead by iterating over each line?